shoop/campaigns/models/context_conditions.py

```diff
@@ -0,0 +1,51 @@
+# This file is part of Shoop.
+#
+# Copyright (c) 2012-2016, Shoop Ltd. All rights reserved.
+#
+# This source code is licensed under the AGPLv3 license found in the
+# LICENSE file in the root directory of this source tree.
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
+from polymorphic.models import PolymorphicModel
+
+from shoop.core.models import ContactGroup
+
+
+class ContextCondition(PolymorphicModel):
+    model = None
+    identifier = "context_condition"
+    name = _("Context Condition")
+    description = _("Context Condition")
+
+    active = models.BooleanField(default=True)
+
+    def matches(self, context):
+        return False
+
+    @property
+    def description(self):
+        extra = "(s)" if self.model else ""
+        return _("Define the required %s%s for this campaign." % (self.name.lower(), extra))
```

Comment:
I would rather say the same (as for CatalogFilter) for this: Provide the message in total for each of the Condition subclasses.

Also: The wording is a bit misleading. Maybe something like "Limit the campaign to members of selected contact groups." would be easier to understand, since the current wording suggests that a contact group is required for *creating* a campaign (rather than it being required for using the campaign)